--- etc/sysconfig/readonly-root.orig	2011-08-16 11:10:34.000000000 -0700
+++ etc/sysconfig/readonly-root	2011-08-16 11:10:53.000000000 -0700
@@ -1,5 +1,5 @@
 # Set to 'yes' to mount the system filesystems read-only.
-READONLY=no
+READONLY=yes
 # Set to 'yes' to mount various temporary state as either tmpfs
 # or on the block device labelled RW_LABEL. Implied by READONLY
 TEMPORARY_STATE=no
--- etc/rc.d/rc.sysinit.orig	2011-08-16 11:11:57.000000000 -0700
+++ etc/rc.d/rc.sysinit	2011-08-16 11:12:07.000000000 -0700
@@ -252,7 +252,7 @@
 	READONLY=no
 fi
 
-if [ "$READONLY" = "yes" -o "$TEMPORARY_STATE" = "yes" ]; then
+if [ "$yes" = "no" ]; then
 
 	mount_empty() {
 		if [ -e "$1" ]; then
@@ -581,7 +581,7 @@
 
 # Clean up /var.
 rm -rf /var/lock/cvs/* /var/run/screen/*
-find /var/lock /var/run ! -type d -exec rm -f {} \;
+find /var/lock/ /var/run/ ! -type d -exec rm -f {} \;
 rm -f /var/lib/rpm/__db* &> /dev/null
 rm -f /var/gdm/.gdmfifo &> /dev/null
 
--- etc/init/rcS.conf.orig	2011-08-16 11:11:14.000000000 -0700
+++ etc/init/rcS.conf	2011-08-16 11:11:29.000000000 -0700
@@ -2,6 +2,10 @@
 #
 # This task runs the old sysv-rc startup scripts.
 
+pre-start script
+	  /sbin/rc.preinit --run --exec "/bin/echo oneSIS: preinit complete"
+end script
+
 start on startup
 
 stop on runlevel
--- etc/init/alces-onesis.conf	1970-01-01 01:00:00.000000000 +0100
+++ etc/init/alces-onesis.conf	2012-03-06 23:07:34.089410282 +0000
@@ -0,0 +1,39 @@
+# rcS - runlevel compatibility
+#
+# This task runs the old sysv-rc startup scripts.
+
+start on startup
+
+stop on runlevel
+
+task
+
+# Note: there can be no previous runlevel here, if we have one it's bad
+# information (we enter rc1 not rcS for maintenance).  Run /etc/rc.d/rc
+# without information so that it defaults to previous=N runlevel=S.
+console output
+pre-start script
+        /sbin/rc.preinit --run --debug --proc --exec="/bin/echo oneSIS: preinit complete"
+	for t in $(cat /proc/cmdline); do
+		case $t in
+			emergency)
+				start rcS-emergency
+				break
+			;;
+		esac
+	done
+end script
+exec /etc/rc.d/rc.sysinit
+post-stop script
+	if [ "$UPSTART_EVENTS" = "startup" ]; then
+		[ -f /etc/inittab ] && runlevel=$(/bin/awk -F ':' '$3 == "initdefault" && $1 !~ "^#" { print $2 }' /etc/inittab)
+		[ -z "$runlevel" ] && runlevel="3"
+		for t in $(cat /proc/cmdline); do
+			case $t in
+				-s|single|S|s) runlevel="S" ;;
+				[1-9])       runlevel="$t" ;;
+			esac
+		done
+		exec telinit $runlevel
+	fi
+end script
